crypto/keys/keybase_base.go

```diff
@@ -167,7 +167,7 @@ func (kb baseKeybase) CreateMnemonic(
 // Derive computes a BIP39 seed from the mnemonic and bip39Passphrase. It creates
 // a private key from the seed using the BIP44 params.
 func (kb baseKeybase) Derive(
-	keyWriter keyWriter, name, mnemonic, bip39Passphrase, encryptPasswd string, params hd.BIP44Params,
+	keyWriter keyWriter, name, mnemonic, bip39Passphrase, encryptPasswd string, params hd.BIP44Params, // nolint:interfacer
```

Comment:
As a suggestion you can change:
```suggestion
	keyWriter keyWriter, name, mnemonic, bip39Passphrase, encryptPasswd string, bip44Params string,
```

and then on L178

```go
retun kb.persistDerivedKey(keyWriter, seed, encryptPasswd, name, bip44Params)
```

Reply:
but then i would have to change where the function is being called, and there are quite a few places so i reverted, i can do then go through and change